libs/community/langchain_community/utilities/pebblo.py

```diff
@@ -711,7 +722,9 @@ def prepare_docs_for_classification(
                     ),
                 }
             )
-        return docs, source_aggregate_size
+            if loader_details["loader"] == "SharePointLoader":
+                loader_details["source_path"] = doc_metadata.get("source_full_url")
```

Comment:
This update statement can be moved outside the loop

Reply:
Needs to be added under iteration, although added flag so that it doesn't get executed in every iteration